httpx/server.go

```diff
@@ -0,0 +1,162 @@
+package httpx
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/go-chi/chi"
+	"github.com/go-chi/chi/middleware"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/rs/cors"
+	"github.com/stellar/go/support/errors"
+	"github.com/stellar/go/support/log"
+)
+
+type ServerMetrics struct {
+	RequestDurationSummary *prometheus.SummaryVec
+}
+
+type TLSConfig struct {
+	CertPath, KeyPath string
+}
+
+type ServerConfig struct {
+	Port               uint16
+	AdminPort          uint16
+	TLSConfig          *TLSConfig
+	PrometheusRegistry *prometheus.Registry
+}
+
+type Server struct {
+	Metrics *ServerMetrics
+
+	server      *http.Server
+	adminServer *http.Server
+
+	tlsConfig          *TLSConfig
```

Comment:
I think we won't ever be handling tls connections here. Like horizon, we would have an ALB or ELB which would terminate the ssl connection

Reply:
I think SDF will use ALB/ELB but not sure about the other organizations. I think it's a good idea to leave this option in case somebody needs it.